### PR TITLE
don't process a group if it has no sources

### PR DESCRIPTION
**Pull request by bbaldino**

### CODE REVIEW

**Comment by bbaldino** [comment]
don't merge this quite yet, want to add some tests

**Comment by bgrozev** [comment]
lgtm. I'm curious what could trigger this condition (an empty group)

**Comment by bbaldino** [comment]
it's caused by this: https://github.com/jitsi/jitsi/blob/smack4-2/src/net/java/sip/communicator/impl/protocol/jabber/extensions/colibri/ColibriBuilder.java#L188

File: src/test/java/org/jitsi/videobridge/xmpp/MediaStreamTrackFactoryTest.java [hunk 1]
```diff
@@ -319,4 +325,57 @@ public void createMediaStreamTracks4()
...
+        MediaStreamTrackReceiver receiver = new MediaStreamTrackReceiver(null);
+
+        MediaStreamTrackDesc[] tracks =
+            MediaStreamTrackFactory.createMediaStreamTracks(receiver,
+                Arrays.asList(
+                    videoSource1),
+                Collections.emptyList());
+
+        assertNotNull(tracks);
+        for (MediaStreamTrackDesc t : tracks) {
```
Comment (bgrozev) [inline src/test/java/org/jitsi/videobridge/xmpp/MediaStreamTrackFactoryTest.java]: Is this intentional?

Comment (bbaldino) [reply src/test/java/org/jitsi/videobridge/xmpp/MediaStreamTrackFactoryTest.java]: shit, no.  will remove

**Pull request merged**